client/src/js/components/userGroups.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { generalUsers } from '../actions/memberActions.js';
+import MemberStore from '../stores/MemberStore.js';
+
+/**
+ * @description - renders Groups Component
+ * @class Groups
+ * @extends {React.Component}
+ */
+export default class Groups extends React.Component {
+  /**
+   * Create a constructor
+   * @constructor
+   * @param {object} props -
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      group: []
+    };
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  componentDidMount() {
+    MemberStore.on('GET_MEMBERS_OF_A_GROUP', this.handleOnClick);
+  }
+
+  componentWillUnmount() {
+    MemberStore.removeListener('GET_MEMBERS_OF_A_GROUP',
+      this.handleOnClick);
+  }
+
+  handleOnClick() {
+    const clickedGroup = MemberStore.allGroupMembers();
+    this.props.getMembers(clickedGroup);
+  }
+
+  /**
+   * @description - render method, React lifecycle method
+   * @returns {Object} Groups component
+   * @Groups
+   */
+  render() {
+    return (
+      <div>
+        <div className="col-md-3">
+          <div className="row chats-row">
+            <div className="col-md-12">
+              <ul className="col-md-10 col-md-offset-1 nav nav-pills
+                nav-stacked grouplist">
+                <li key="general"
+                  value="general" name="general"
+                  onClick={() => generalUsers()}>
+                  <Link to="#">general</Link></li>
+                {this.props.grouplist}
+              </ul>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+
+// props validation
+Groups.propTypes = {
+  grouplist: PropTypes.array,
```

Comment:
You should look for a proper way to validate arrays. Your eslint should have complaint